app/src/main/java/com/framgia/youtube_background_01/screen/home/HomeFragment.java

```diff
@@ -1,6 +1,96 @@
+import com.framgia.youtube_background_01.data.model.Video;
+import com.framgia.youtube_background_01.data.repository.VideoRepository;
+import com.framgia.youtube_background_01.data.source.local.VideoLocalDataSource;
+import com.framgia.youtube_background_01.data.source.remote.VideoRemoteDataSource;
+import com.framgia.youtube_background_01.screen.home.adapter.PlaylistAdapter;
+import com.framgia.youtube_background_01.screen.home.adapter.TrendingAdapter;
+import java.util.List;
 
-public class HomeFragment extends Fragment {
+public class HomeFragment extends Fragment implements HomeContract.View {
+    private TrendingAdapter mTrendingAdapter;
+    private PlaylistAdapter mPlaylistAdapter;
+    private List<Video> mVideosTrending;
+    private List<PlayList> mPlayLists;
+
+    public static HomeFragment newInstance() {
+        HomeFragment homeFragment = new HomeFragment();
+        Bundle args = new Bundle();
+        homeFragment.setArguments(args);
+        return homeFragment;
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+            @Nullable Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+        View view = inflater.inflate(R.layout.fragment_home, container, false);
+        initView(view);
+        initData();
+        return view;
+    }
+
+    private void initView(View view) {
+        RecyclerView trendingRecyclerView = view.findViewById(R.id.recyclerViewTrending);
+        trendingRecyclerView.setHasFixedSize(true);
+        trendingRecyclerView.setNestedScrollingEnabled(false);
+        mTrendingAdapter = new TrendingAdapter(getContext());
+        trendingRecyclerView.setAdapter(mTrendingAdapter);
+        RecyclerView playlistRecyclerView = view.findViewById(R.id.recyclerViewPlaylist);
+        playlistRecyclerView.setLayoutManager(new GridLayoutManager(getContext(), 2));
+        playlistRecyclerView.setHasFixedSize(true);
+        playlistRecyclerView.setNestedScrollingEnabled(false);
+        mPlaylistAdapter = new PlaylistAdapter(getContext());
+        playlistRecyclerView.setAdapter(mPlaylistAdapter);
+    }
+
+    private void initData() {
+        VideoRemoteDataSource trackRemoteDataSource = VideoRemoteDataSource.getInstance();
+        VideoLocalDataSource trackLocalDataSource = VideoLocalDataSource.getInstance();
+        VideoRepository repository =
+                VideoRepository.getInstance(trackLocalDataSource, trackRemoteDataSource);
+        HomeContract.Presenter presenter = new HomePresenter(repository);
+        presenter.setView(this);
+        presenter.getGetTrendingVideo();
+        presenter.getPlayList();
+    }
+
+    @Override
+    public void onGetPlaylistSuccess(List<PlayList> lists) {
+        if (lists != null) {
+            mPlayLists = lists;
+            mPlaylistAdapter.updateData(mPlayLists);
```

Comment:
e tạo thêm mPlayLists có chức năng gì vậy nhỉ?
```suggestion
            mPlaylistAdapter.updateData(lists);
```